argus/backend/service/github_service.py

```diff
@@ -0,0 +1,232 @@
+import re
+import logging
+from collections import defaultdict
+from datetime import UTC, datetime
+from functools import reduce
+from uuid import UUID
+from flask import current_app, g
+from github import Github, Auth
+
+from argus.backend.models.runtime_store import RuntimeStore
+from argus.backend.models.web import ArgusEventTypes, ArgusTest, ArgusUserView
+from argus.backend.models.github_issue import GithubIssue, IssueAssignee, IssueLink, IssueLabel
+from argus.backend.plugins.core import PluginInfoBase
+from argus.backend.plugins.loader import AVAILABLE_PLUGINS
+from argus.backend.service.event_service import EventService
+from argus.backend.util.common import chunk
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GithubService:
```

Comment:
general question, how will it behave when someone moves the issue?

Reply:
Badly. I contemplate also syncing the github internal issue id to alleviate this problem (also would bulletproof comparing local issue with remote), but I'm not sure what happens on the API side in a situation where you request a repo update and there's an issue that has been moved to an untracked repo.

Reply:
badly - means breaking the process, or just not updating?
if issue disappears from the repo, we could add a label `moved`, without pointing where if it's hard to get (anyway, rare situation).